src/RazorpayClient.cs

```diff
@@ -1,233 +1,248 @@
-
-        public Invoice Invoice
-        {
-            get
-            {
-                if (invoice == null)
-                {
-                    invoice = new Invoice();
-                }
-                return invoice;
-            }
-        }
-
-        public Card Card
-        {
-            get
-            {
-                if (card == null)
-                {
-                    card = new Card();
-                }
-                return card;
-            }
-        }
-
-        public Transfer Transfer
-        {
-            get
-            {
-                if (transfer == null)
-                {
-                    transfer = new Transfer();
-                }
-                return transfer;
-            }
-        }
-
-        public Addon Addon
-        {
-            get
-            {
-                if (addon == null)
-                {
-                    addon = new Addon();
-                }
-                return addon;
-            }
-        }
-
-        public Plan Plan
-        {
-            get
-            {
-                if (plan == null)
-                {
-                    plan = new Plan();
-                }
-                return plan;
-            }
-        }
-
-        public Subscription Subscription
-        {
-            get
-            {
-                if (subscription == null)
-                {
-                    subscription = new Subscription();
-                }
-                return subscription;
-            }
-        }
-
-        public VirtualAccount VirtualAccount
-        {
-            get
-            {
-                if (virtualaccount == null)
-                {
-                    virtualaccount = new VirtualAccount();
-                }
-                return virtualaccount;
-            }
-        }
-    }
-}
+using System.Collections.Generic;
+
+namespace Razorpay.Api
+{
+    public class RazorpayClient
+    {
+        protected const string version = "3.0.0";
+
+        protected static List<Dictionary<string, string>> appsDetails = new List<Dictionary<string, string>>();
+        protected static Dictionary<string, string> headers = new Dictionary<string, string>();
+
+        private static string key = null;
+        private static string secret = null;
+        private static string baseUrl = "https://api.razorpay.com/v1/";
+        private Payment payment = null;
+        private Order order = null;
+        private Refund refund = null;
+        private Customer customer = null;
+        private Invoice invoice = null;
+        private Card card = null;
+        private Transfer transfer = null;
+        private Addon addon = null;
+        private Plan plan = null;
+        private Subscription subscription = null;
+        private VirtualAccount virtualaccount = null;
+
+        
+        public RazorpayClient(string key, string secret)
+        {
+            RazorpayClient.Key = key;
+            RazorpayClient.Secret = secret;
```

Comment:
initialize BaseUrl as well

Reply:
Already initialized it in another constructor.